solutions/USACO dec18-bucketlist.mdx

```diff
@@ -41,3 +41,62 @@ file_out.close()
 </PySection>
 
 </LanguageSection>
+
+## Solution 2
+
+Similar to the second solution in the official editorial. 
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>  
+#include <cstdio>
+using namespace std;
+int event[1001]; // declaring the array globally to initialize all the elements to 0
+int main() {
+	freopen("blist.in", "r", stdin);
+	freopen("blist.out", "w", stdout); 
+	int maxtime = 1000;
+	int n; cin >> n;
+		
+		
+	for (int i = 0; i < n; i++) { // iterating through every cow
+		int start; cin >> start; 
+		int end; cin >> end;
+		int amt; cin >> amt;
+		// at time start, cow i needs to be milked, so 
+		// we'll need amt additional buckets (hence adding amt).
+		// at time end, cow i is done milking, so we can free up 
+		// amt buckets (hence subtracting amt)
```

Comment:
maybe you could put the first two lines above `event[start] += amt;` and the last two lines above `event[end] -= amt;` ?

Reply:
i think that would mess up the flow

Reply:
gonna do it then if lunchbox agrees as well

Reply:
> maybe you could put the first two lines above `event[start] += amt;` and the last two lines above `event[end] -= amt;` ?

i agree